server/src/main/java/org/solarbank/server/service/NasaClient.java

```diff
@@ -0,0 +1,55 @@
+package org.solarbank.server.service;
+
+import java.time.Duration;
+import java.util.Map;
+import org.solarbank.server.ErrorMessage;
+import org.solarbank.server.dto.Location;
+import org.solarbank.server.dto.NasaResponse;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+import reactor.util.retry.Retry;
+
+@Service
+public class NasaClient {
+    private final WebClient webClient;
+    private static final String allSkyParameter = "ALLSKY_SFC_SW_DWN";
+    private static final int MAX_RETRY_ATTEMPTS = 3;
+    private static final int DELAY_MILLIS = 100;
+
+    public NasaClient(@Value("${nasa.api.url}") String baseUrl) {
+        this.webClient = WebClient.create(baseUrl);
+    }
+
+    public Map<String, Double> getNasaData(Location location) {
+        NasaResponse nasaResponse = webClient.get()
+                .uri(uriBuilder -> uriBuilder
+                        .path("/point")
+                        .queryParam("parameters", allSkyParameter)
+                        .queryParam("community", "RE")
+                        .queryParam("longitude", location.getLongitude())
+                        .queryParam("latitude", location.getLatitude())
+                        .queryParam("start", "2012")
+                        .queryParam("end", "2021")
```

Comment:
start and end are set as the monthly Nasa data only goes up to 2022, so just use a set 10 years of data to calculate average.